velox/connectors/hive/iceberg/TransformEvaluator.h

```diff
@@ -0,0 +1,99 @@
+#include "velox/expression/Expr.h"
+
+namespace facebook::velox::connector::hive::iceberg {
+
+/// Utility class for converting Iceberg partition specification to Velox
+/// expressions.
+///
+/// Provides static methods to build expression trees from Iceberg partition
+/// specification. These expressions can be compiled and evaluated using
+/// Velox's expression evaluation framework.
+class TransformExprBuilder {
+ public:
+  /// Converts a single partition field to a typed expression.
+  ///
+  /// Builds an expression tree for one partition transform. Identity transforms
+  /// become FieldAccessTypedExpr, while other transforms (bucket, truncate,
+  /// year, month, day, hour) become CallTypedExpr with appropriate function
+  /// names and parameters.
+  ///
+  /// @param field Partition field containing transform type, source column
+  /// type, and optional parameter (e.g., bucket count, truncate width).
+  /// @param inputFieldName Name of the source column in the input RowVector.
+  /// @param inputType Type of the source column.
+  /// @return Typed expression representing the transform.
+  static core::TypedExprPtr toExpression(
+      const IcebergPartitionSpec::Field& field,
+      const std::string& inputFieldName,
+      const TypePtr& inputType);
+
+  /// Converts partition specification to a list of typed expressions.
+  ///
+  /// Builds expression trees for all partition transforms defined in the
+  /// partition spec. Each transform is converted to either a
+  /// FieldAccessTypedExpr (for identity transforms) or a CallTypedExpr (for
+  /// other transforms).
+  ///
+  /// @param partitionSpec Iceberg partition specification containing transform
+  /// definitions for each partition field.
+  /// @param partitionChannels Column indices (0-based) in the input RowVector
+  /// that correspond to each partition field. Must have the same size as
+  /// partitionSpec->fields.
+  /// @param inputType The row type of the input data, used to extract column
+  /// names and types for building expressions.
+  /// @return Vector of typed expressions, one for each partition field.
+  static std::vector<core::TypedExprPtr> toExpressions(
+      const IcebergPartitionSpecPtr& partitionSpec,
+      const std::vector<column_index_t>& partitionChannels,
+      const RowTypePtr& inputType);
+};
+
+/// Evaluates multiple transforms efficiently using batch expression evaluation.
+class TransformEvaluator {
+ public:
+  /// Creates an evaluator with the given connector query context.
+  ///
+  /// @param connectorQueryCtx Connector query context providing access to the
+  /// expression evaluator (for compilation and evaluation) and memory pool.
+  /// Must remain valid for the lifetime of this TransformEvaluator.
+  TransformEvaluator(const ConnectorQueryCtx* connectorQueryCtx);
+
+  /// Evaluates all expressions on the input data in a single pass.
+  /// Compiles the expressions into an ExprSet and evaluates them together for
```

Comment:
I assume we don't want to re-compile the expressions for each batch of data. Instead, we'd compile them once, but evaluate many times. See FilterProject.cpp for reference.